src/Domain/CoordinateMaps/CoordinateMap.hpp

```diff
@@ -257,47 +342,93 @@ class CoordinateMap
 // CoordinateMap definitions
 ////////////////////////////////////////////////////////////////
 
+// define type-trait to check for time-dependent mapping
+namespace CoordinateMap_detail {
+template <typename T>
+using is_map_callable_t =
+    tt::is_callable_t<T, std::array<std::decay_t<T>, std::decay_t<T>::dim>,
+                      double, std::unordered_map<std::string, FunctionOfTime&>>;
+}  // namespace CoordinateMap_detail
+
 template <typename SourceFrame, typename TargetFrame, typename... Maps>
 constexpr CoordinateMap<SourceFrame, TargetFrame, Maps...>::CoordinateMap(
     Maps... maps)
     : maps_(std::move(maps)...) {}
 
 template <typename SourceFrame, typename TargetFrame, typename... Maps>
-template <typename T>
+template <typename T, size_t... Is>
 constexpr SPECTRE_ALWAYS_INLINE tnsr::I<
     T, CoordinateMap<SourceFrame, TargetFrame, Maps...>::dim, TargetFrame>
 CoordinateMap<SourceFrame, TargetFrame, Maps...>::call_impl(
-    tnsr::I<T, dim, SourceFrame>&& source_point) const {
+    tnsr::I<T, dim, SourceFrame>&& source_point, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& f_of_t_list,
+    std::index_sequence<Is...>) const noexcept {
```

Comment:
I've always named these metaprogramming useful variables `/*meta*/` to indicate they purely for metaprogramming.